Source/WebKit/UIProcess/WebFrameProxy.h

```diff
@@ -187,6 +187,9 @@ class WebFrameProxy : public API::ObjectImpl<API::Object::Type::Frame>, public C
     TraversalResult traverseNext(CanWrap) const;
     TraversalResult traversePrevious(CanWrap);
 
+    void setHasPendingBackForwardItem() { m_hasPendingBackForwardItem = true; }
+    bool takeHasPendingBackForwardItem() { return std::exchange(m_hasPendingBackForwardItem, false); }
```

Comment:
I think it would be better to just have setHasPendingBackForwardItem(bool) because you're not really taking anything and the return value of take is not currently used.